### ISSUE TITLE
Penetration Resistance not being accounted in build comparison

### ISSUE THREAD

**Issue by jcari-dev** [tyopoyt/epic7-damage-calc#179]
Currently the values of penetration resistance are not being accounted for the comparison builds.

Steps to reproduce:

1. Select MG Lilias add some values and change caster and target max hp, etc. Sample build below:
 4000 atk,
 250 cd, 
 caster hp 8k, 
 target hp 35k, 
 target def 1k, 
 caster w attack buff 
 and save the build.

![image](https://github.com/user-attachments/assets/bd54c6ee-5549-4fe6-b632-9e7c38086128)

2. Change the penetration resistance values (to any value, e.g., 30%)

![image](https://github.com/user-attachments/assets/8beb6c10-816e-4edf-ac99-f509c59db31f)

_Note the damage values for critical hit in the calculator have changed (18242 -> 9121)_

3. Attempt to save the build:

![image](https://github.com/user-attachments/assets/9be60fc3-c655-4c0d-a37d-ee8baae963b5)

_Notice that the calculator doesn't recognize the change in the values, for some reason it thinks it is the same build_.

After this point the build would not be saved.

4. Change any value within the calculator in order to change the actual damage values, for example, target defense (1000 -> 1001):
![image](https://github.com/user-attachments/assets/c615af19-f00d-4683-b970-f98e8866fae6)

_Now the calculator recognizes the changes in values, and accepts the build for comparison (build 2, with 1001 also contains 30% pen resist)_
![image](https://github.com/user-attachments/assets/541406bf-0294-4ae7-97cc-d2c0c14dc0d8)

Let me know if there's anything I can help on! Thank you for continuing to develop this amazing tool.

**Comment by tyopoyt** [comment]
Hi, thanks for the report with detailed steps and pics! It looks like what's happening here is that none of the values that are used to generate the default name of the build have changed so it's wanting to overwrite the saved build with the same name.  If you change the name of the build at all (you can set build names to whatever you want on the save screen, don't have to stick with the default one) it should save a separate build rather than overwriting the previous.

There are a _ton_ of things that could possible be changed and affect the damage results so I don't think it makes much sense to try and put everything at all into the build name.  I definitely don't want to change the behavior that saving a build with the same name overwrites a previous build with the same name because then it would be confusing to know which build is which in the list.

If you have any suggestions about how this can be handled differently I'm open to that.  Maybe you didn't notice the build name is editable and you have an idea what would've helped you notice?



**TL;DR** I would recommend adding something like `(50% Pen res)` to the end of the build with pen res before saving.
